test/declarative.jl

```diff
@@ -3,122 +3,114 @@ using JSON, BinaryBuilder, Test
 import BinaryBuilder.BinaryBuilderBase: sourcify, dependencify, major, minor, patch, version
 
 @testset "Meta JSON" begin
-    mktempdir() do tmpdir
+    meta_json_buff = IOBuffer()
+
+    # Run autobuild() a few times to generate a moderately complex `meta.json`:
+    dict = get_meta_json(
+        "libfoo",
+        v"1.0.0",
+        [FileSource("https://julialang.org", "123123"), DirectorySource("./bundled")],
+        "exit 1",
+        [Linux(:x86_64)],
+        Product[LibraryProduct("libfoo", :libfoo), FrameworkProduct("fooFramework", :libfooFramework)],
+        [Dependency("Zlib_jll")];
+    )
+    println(meta_json_buff, JSON.json(dict))
```

Comment:
Isn't there anything interesting we can test on the dictionary?

Reply:
I am sure one could. For example, by comparing it with the decoded JSON data in `objs` down below -- although that'd essentially test the JSON encoder/decoder; not sure if that's so useful?

However, I deliberately tried to restrict the changes in this PR to refactoring, not adding new stuff.

Reply:
Sure, but there was a minimal test before :wink:

Reply:
OK, I must have missed that, what test was there that I accidentally removed? It still serializes and deserializes the dict to JSON and back as before.